### PR TITLE
perf(policies): if no trafficRoute don't read old policies

### PR DESCRIPTION
**Pull request by lahabana**
- Old policies are not usable if you don't have trafficRoute
- Therefore we don't put them in the MeshContext
- Refactored things to make clear what is used by old policies and that we shouldn't use anymore

This is a first step to get away from having proxy generation be in O(number of dataplanes).

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues --
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/core/xds/inspect/attachments_test.go [hunk 1]
```diff
@@ -301,12 +302,12 @@ var _ = Describe("GroupByAttachment", func() {
...
 					"backend": &core_mesh.RetryResource{Meta: meta1},
 				},
-				Dynamic: map[core_model.ResourceType]core_xds.TypedMatchingPolicies{
-					core_mesh.TrafficLogType: {
-						ServicePolicies: map[core_xds.ServiceName][]core_model.Resource{
-							"redis": {
-								&core_mesh.TrafficLogResource{Meta: meta6},
-							},
+			},
+			policyPlugins: core_xds.PluginOriginatedPolicies{
```
Comment (lukidzi) [inline pkg/core/xds/inspect/attachments_test.go:306]: Just a bit confusing that we are using `PluginOriginatedPolicies` with old policies

**Review by lukidzi — APPROVED** [review]

File: pkg/core/xds/inspect/attachments_test.go (hunk 1 above)
Comment (lahabana) [reply pkg/core/xds/inspect/attachments_test.go:306]: Are we?

File: pkg/api-server/inspect_endpoints_test.go [hunk 2]
```diff
@@ -60,6 +60,22 @@ func serviceSelector(name, protocol string) *mesh_proto.Selector {
 	}
 }
 
+func defaultTrafficRoute(mesh string) *core_mesh.TrafficRouteResource {
```
Comment (slonka) [inline pkg/api-server/inspect_endpoints_test.go:63]: nitpick: could've been a reusable `route-all-default` test resource - but we're not going to develop old policies more so I think this can stay.

File: pkg/core/xds/inspect/attachments_test.go (hunk 1 above)
Comment (slonka) [reply pkg/core/xds/inspect/attachments_test.go:306]: We're using both right? `GroupByAttachment` takes both `PluginOriginatedPolicies` (new policies) and `MatchedPolicies` (old policies) and then in (e.g.) `getInboundMatchedPolicies` we combine them.

File: pkg/plugins/policies/meshfaultinjection/plugin/v1alpha1/plugin.go [hunk 3]
```diff
@@ -71,9 +71,6 @@ func applyToInbounds(
 	for _, inbound := range proxy.Dataplane.Spec.GetNetworking().GetInbound() {
 		iface := proxy.Dataplane.Spec.Networking.ToInboundInterface(inbound)
 		protocol := core_mesh.ParseProtocol(inbound.GetProtocol())
-		if _, exists := proxy.Policies.FaultInjections[iface]; exists {
```
Comment (slonka) [inline pkg/plugins/policies/meshfaultinjection/plugin/v1alpha1/plugin.go:74]: why was this here in the first place @lukidzi ? why do we not need it now @lahabana ?

**Review by slonka — APPROVED** [review]
Just a couple of Qs. This is a nice addition but have we seen this being a perf bottleneck anywhere?

**Comment by lahabana** [comment]
I'm going to close this. I think the right option is to potentially not install old policies in this case.
